daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-cloudevents.md

```diff
@@ -102,6 +117,10 @@ Invoke-RestMethod -Method Post -ContentType 'application/cloudevents+json' -Body
 
 {{< /tabs >}}
 
+## Event deduplication
+
+When using cloud events created by Dapr, the envelope contains an `id` field which can be used by the app to perform message deduplication. Dapr does not handle deduplication automatically. Dapr supports using message brokers that natively enable message deduplication.
```

Comment:
When is says "Dapr supports using message brokers that natively enable message deduplication" should this should say that deduplication can only be done by the broker? Of the 20 brokers Dapr is integrated with, they do not all have deduplication (e.g. In Memory https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-inmemory/)

Reply:
No, because deduplication can also be done manually via the app. In effect, the text here says that:

1. An app can perform deduplication
2. A message broker can perform deduplication if supported
3. Dapr does not handle deduplication